c_src/myhtml_worker.c

```diff
@@ -256,140 +257,116 @@ read_parse_flags(ETERM* list)
 
   return parse_flags;
 }
-
-ETERM*
-build_tree(prefab_t* prefab, myhtml_tree_t* tree, myhtml_tree_node_t* node, unsigned char* parse_flags)
+ETERM* build_tree(prefab_t* prefab, myhtml_tree_t* tree, myhtml_tree_node_t* node, unsigned char* parse_flags)
 {
   ETERM* result;
-  myhtml_tag_id_t tag_id = myhtml_node_tag_id(node);
-  myhtml_namespace_t tag_ns = myhtml_node_namespace(node);
+  myhtml_tree_node_t* prev_node = NULL;
 
-  if (tag_id == MyHTML_TAG__TEXT)
-  {
-    size_t text_len;
-    const char* node_text = myhtml_node_text(node, &text_len);
-    result = erl_mk_binary(node_text, text_len);
-  }
-  else if (tag_id == MyHTML_TAG__COMMENT)
-  {
-    size_t comment_len;
-    const char* node_comment = myhtml_node_text(node, &comment_len);
-    ETERM* comment = erl_mk_binary(node_comment, comment_len);
-
-    if (*parse_flags & FLAG_COMMENT_TUPLE3)
-    {
-      /* ETERM* tuple3[] = {prefab->atom_comment, prefab->empty_list, comment}; */
-      /* result = erl_mk_tuple(tuple3, 3); */
-      result = erl_format("{comment, [], ~w}", comment);
-    }
-    else
-    {
-      /* ETERM* tuple2[] = {prefab->atom_comment, comment}; */
-      /* result = erl_mk_tuple(tuple2, 2); */
-      result = erl_format("{comment, ~w}", comment);
-    }
-  }
-  else
-  {
-    ETERM* tag;
-    ETERM* attrs;
+  tstack stack;
+  tstack_init(&stack, 30);
```

Comment:
I went through this change with a fellow programmer of mine, and now after following through and understanding, I think its wayyy smarter to use a stack here, than my previous implementation.

Makes perfect sense and should be much safer.